src/fswAlgorithms/stateEstimation/thrustCMEstimation/thrustCMEstimation.cpp

```diff
@@ -72,5 +72,90 @@ void ThrustCMEstimation::Reset(uint64_t CurrentSimNanos)
  */
 void ThrustCMEstimation::UpdateState(uint64_t CurrentSimNanos)
 {
-    
+    /*! create output message buffers */
+    VehicleConfigMsgPayload vehConfigOutBuffer;
+    CMEstDataMsgPayload cmEstDataBuffer;
+
+    /*! read and allocate the thrustConfigMsg */
+    THRConfigMsgPayload thrConfigBuffer = this->thrusterConfigBInMsg();
+
+    /*! compute thruster information in B-frame coordinates */
+    Eigen::Vector3d r_TB_B = cArray2EigenVector3d(thrConfigBuffer.rThrust_B);
+    Eigen::Vector3d T_B = thrConfigBuffer.maxThrust * cArray2EigenVector3d(thrConfigBuffer.tHatThrust_B);
+
+    /*! compute error w.r.t. target attitude */
+    AttGuidMsgPayload attGuidBuffer = this->attGuidInMsg();
+    Eigen::Vector3d sigma_BR   = cArray2EigenVector3d(attGuidBuffer.sigma_BR);
+    Eigen::Vector3d omega_BR_B = cArray2EigenVector3d(attGuidBuffer.omega_BR_B);
+    double attError = pow(sigma_BR.squaredNorm() + omega_BR_B.squaredNorm(), 0.5);
+
+    /*! read commanded torque msg */
+    CmdTorqueBodyMsgPayload cmdTorqueBuffer = this->intFeedbackTorqueInMsg();
+    Eigen::Vector3d L_B = -cArray2EigenVector3d(cmdTorqueBuffer.torqueRequestBody);
+
+    Eigen::Vector3d y;       // measurement
+    Eigen::Vector3d preFit;  // pre-fit residual
+    Eigen::Vector3d postFit; // post-fit residual
+    Eigen::Matrix3d H;       // linear model
+    Eigen::Matrix3d K;       // kalman gain
+    Eigen::Matrix3d S;
+
+    /*! assign preFit and postFit residuals to NaN, rewrite them in case of measurement update */
+    preFit  = {nan("1"), nan("1"), nan("1")};
```

Comment:
:large_orange_diamond: Can you help me understand why a nan is chosen here as opposed to something else? I don't have a strong opinion and so would like to hear yours. We can huddle about this one.